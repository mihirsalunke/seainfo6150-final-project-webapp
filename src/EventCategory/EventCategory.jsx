import React from "react";
import styles from "./EventCategory.module.css";
import PropTypes from "prop-types";

const EventCategory = (props) => {
    return (
        <div className={styles.category}>
            {props.category.categoryName}
        </div>
    );
};

EventCategory.propTypes = {
    category: PropTypes.object.isRequired,
};

export default EventCategory;