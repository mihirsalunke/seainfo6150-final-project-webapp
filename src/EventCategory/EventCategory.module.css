@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nerko+One&display=swap");

.category {
    background-image: linear-gradient(to right, rgb(202, 208, 211), rgb(230, 238, 243) 70%);
    
    border-radius: 10px;
    border: 2px solid black;
    color: rgb(22, 28, 39);
    padding: 10px;
    font-size: 1.5em;
    font-family: Rowdies, sans-serif;
    margin: 0 0 20px;
    width: 5em;
    text-align: center;
    vertical-align: middle;

    transition: transform .25s ease-out;
}

.category:hover {
    transform: translateY(2px);
}

.category:active {
    transform: translateY(1px);
}

  @media screen and (min-width: 525px) {
    .category {
        padding: 10px;
        font-size: 0.7em;
        margin: 0 0 20px;
        width: 2.5em;
    }
  }
  
  @media screen and (min-width: 800px) {
    .category {
        padding: 10px;
        font-size: 1.5em;
        margin: 0 0 20px;
        width: 3em;
    }
  }
 
  
  @media screen and (min-width: 1150px) {
    .category {
        padding: 10px;
        font-size: 2em;
        margin: 0 0 20px;
        width: 5em;
    }
  }