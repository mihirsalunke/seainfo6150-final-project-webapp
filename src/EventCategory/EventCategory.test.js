import React from "react";
import EventCategory from "./EventCategory.jsx";
import { render } from "@testing-library/react";

const category = {
    "categoryName": "Sports",
    "categoryId": "1"
};

describe("EventCategory tests", () => {
    it("renders correctly", () => {
      const { container } = render(<EventCategory category={category} />);
      expect(container).toMatchSnapshot();
    });
});