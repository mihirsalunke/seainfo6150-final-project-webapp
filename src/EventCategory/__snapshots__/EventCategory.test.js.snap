// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCategory tests renders correctly 1`] = `
<div>
  <div
    class="category"
  >
    Sports
  </div>
</div>
`;
